pkg/obfuscator/tracker.go

```diff
@@ -0,0 +1,84 @@
+package obfuscator
+
+import (
+	"sync"
+
+	"k8s.io/klog/v2"
+)
+
+type GenerateReplacement func(string) string
+
+// ReplacementTracker is used to track and generate replacements used by obfuscators
+type ReplacementTracker interface {
+	// Report returns a mapping of strings which were replaced.
+	Report() map[string]string
+
+	// AddReplacement will add a replacement along with its original string to the report.
+	// If there is an existing value that does not match the given replacement, it will exit with a non-zero status.
+	AddReplacement(original string, replacement string)
+
+	// GetReplacement returns the previously used replacement if already set. If the replacement is not present then it
+	// uses the GenerateReplacement function to generate a replacement. If the generator is nil then it returns an empty
+	// string. The original parameter must be used for lookup and the key parameter to generate the replacement.
+	GetReplacement(original string, key string, generator GenerateReplacement) string
+
+	// Initialize initializes the tracker with some existing replacements. It should be called only once and before
+	// the first use of GetReplacement or AddReplacement
+	Initialize(replacements map[string]string)
```

Comment:
super nit pick, add that to the top of the type, so people know that this needs to be initialized first.

Reply:
Done